core/backend/src/annotations/LeaderGeometry.ts

```diff
@@ -78,26 +78,116 @@ export function appendLeadersToBuilder(builder: ElementGeometry.Builder, leaders
 
     leaderLinePoints.push(attachmentPoint)
 
-    result = result && builder.appendGeometryQuery(LineString3d.create(leaderLinePoints));
-
-    // Terminator geometry
     const terminatorDirection = Vector3d.createStartEnd(
       leaderLinePoints[0], leaderLinePoints[1]
     ).normalize();
-
-    const termY = terminatorDirection?.unitCrossProduct(Vector3d.unitZ());
-    if (!termY || !terminatorDirection) continue; // Assuming leaders without terminators is a valid case.
-    const terminatorHeight = leaderStyle.leader.terminatorHeightFactor * scaledBlockTextHeight;
     const terminatorWidth = leaderStyle.leader.terminatorWidthFactor * scaledBlockTextHeight;
-    const basePoint = leader.startPoint.plusScaled(terminatorDirection, terminatorWidth);
-    const termPointA = basePoint.plusScaled(termY, terminatorHeight);
-    const termPointB = basePoint.plusScaled(termY.negate(), terminatorHeight);
-    result = result && builder.appendGeometryQuery(LineString3d.create([termPointA, leader.startPoint, termPointB]));
+    // Truncate the first segment of the leader lines to account for the arrowhead size when closedArrow (hollow triangle) terminatorShape is used.
+    if (leaderStyle.leader.terminatorShape === "closedArrow") {
+      if (terminatorDirection)
+        leaderLinePoints[0] = leaderLinePoints[0].plusScaled(terminatorDirection, terminatorWidth);
+    }
+
+    result = result && builder.appendGeometryQuery(LineString3d.create(leaderLinePoints));
+
+    // Terminator geometry
+    if (!terminatorDirection) continue; // Assuming leaders without terminators is a valid case.
```

Comment:
From what I can tell, you *don't* permit leaders without terminators. `TerminatorShape` has no "none" option and when you migrate pre-terminator text style settings you force them to use the default terminator shape. Did I overlook something? If not, is it true that we do not want to support leaders without terminators?